registry/storage/driver/s3/s3.go

```diff
@@ -587,6 +587,11 @@ func (d *driver) List(path string) ([]string, error) {
 	if path != "/" && path[len(path)-1] != '/' {
 		path = path + "/"
 	}
+	prefix := ""
```

Comment:
Absolutely not. That converts to paths to be used for s3 calls, whereas this prefix is used for what needs to be returned to by this driver.